dotcom-rendering/src/web/components/KeyEventsCarousel.importable.tsx

```diff
@@ -0,0 +1,204 @@
+	background-color: ${neutral[0]};
+	bottom: 4px;
+	&:active,
+	&:hover {
+		outline: none;
+		background-color: ${brandAlt[400]};
+		svg {
+			fill: ${neutral[7]};
+		}
+	}
+
+	&:focus {
+		outline: none;
+	}
+
+	svg {
+		fill: ${neutral[100]};
+		height: 34px;
+	}
+`;
+
+const disabledButtonStyles = css`
+	background-color: ${neutral[60]};
+	cursor: default;
+
+	&:hover,
+	&:focus {
+		background-color: ${neutral[60]};
+
+		svg {
+			fill: ${neutral[100]};
+		}
+	}
+`;
+
+export const KeyEventsCarousel = ({
+	keyEvents,
+	filterKeyEvents,
+	format,
+}: Props) => {
+	const [activeIndex, setActiveIndex] = useState(0);
+	const [carousel, setCarousel] = useState<HTMLElement>();
+	const [cards, setCards] = useState<Element[]>();
+	const cardWidth = 200;
+
+	// we only want to count cards fully in view so we round down.
+	const cardsOnScreen = carousel
+		? Math.floor(carousel.clientWidth / cardWidth)
+		: 0;
+
+	const isFirstCard = activeIndex === cardsOnScreen;
+	const isLastCard = activeIndex === keyEvents.length - 1;
+
+	useEffect(() => {
+		const carouselElement = document.getElementById('key-event-carousel');
+		const cardsArray = Array.from(
+			document.querySelectorAll('#key-event-card'),
+		);
+		if (carouselElement) setCarousel(carouselElement);
+		if (cardsArray) setCards(cardsArray);
+	}, []);
+
+	useEffect(() => {
+		const handleIntersect = (entries: IntersectionObserverEntry[]) => {
+			const entry = entries.find((e) => e.isIntersecting);
+			const index = cards?.findIndex((e) => e === entry?.target);
+			if (index && index !== -1) setActiveIndex(index);
+		};
+
+		const observer = new window.IntersectionObserver(handleIntersect, {
+			root: carousel,
+			rootMargin: '0% 0% 0% -100%',
+		});
+
+		cards?.forEach((el) => {
+			observer.observe(el);
+		});
+
+		return () => observer.disconnect();
+	}, [activeIndex, cards, carousel]);
+
+	const goPrevious = () => {
+		if (carousel && !isFirstCard) carousel.scrollLeft -= cardWidth;
+	};
+
+	const goNext = () => {
+		if (carousel && !isLastCard) carousel.scrollLeft += cardWidth;
+	};
+
+	const transformedKeyEvents = keyEvents
+		.filter((keyEvent) => {
+			return keyEvent.title && keyEvent.blockFirstPublished;
+		})
+		.map((keyEvent) => {
+			return {
+				text: keyEvent.title || '', // We fallback to '' here purely to keep ts happy
+				url: `?filterKeyEvents=${filterKeyEvents}&page=with:block-${keyEvent.id}#block-${keyEvent.id}`,
+				date: new Date(keyEvent.blockFirstPublished || ''), // We fallback to '' here purely to keep ts happy
+			};
+		});
+	return (
+		<div id="key-event-carousel" css={carouselStyles}>
+			<span css={containerStyles}>
+				{transformedKeyEvents.map((keyEvent) => {
+					return (
+						<KeyEventCard
+							text={keyEvent.text}
+							url={keyEvent.url}
+							date={keyEvent.date}
+							format={format}
+						/>
+					);
+				})}
+			</span>
+			<Hide until="desktop">
+				{keyEvents.length > 6 && (
+					<span css={buttonContainerStyles}>
+						<button
```

Comment:
I suspect there's a good reason for this - I vaguely recall this was discussed before so sorry if I'm repeating myself - but just to be sure, are we able to use `Button` from source here?

Reply:
Ah good spot, I think we should be fine to use Source here.